ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityMediationAdapter.java

```diff
@@ -236,8 +193,48 @@ public void loadRewardedAd(MediationRewardedAdConfiguration mediationRewardedAdC
 
         mMediationAdLoadCallback = mediationAdLoadCallback;
 
-        UnitySingleton.getInstance().initializeUnityAds((Activity) context, gameID);
-        UnitySingleton.getInstance().loadAd(mUnityAdapterRewardedAdDelegate);
+        UnityInitializer.getInstance().initializeUnityAds((Activity) context, gameID, new IUnityAdsInitializationListener() {
+            @Override
+            public void onInitializationComplete() {
+                Log.d(UnityAdapter.TAG, "Unity Ads successfully initialized");
+                loadRewardedAd();
+            }
+
+            @Override
+            public void onInitializationFailed(UnityAds.UnityAdsInitializationError unityAdsInitializationError, String s) {
+                Log.e(UnityAdapter.TAG, "Unity Ads initialization failed: [" + unityAdsInitializationError + "] " + s);
+            }
+        });
```

Comment:
Does Facebook adapter has the same code?

Reply:
it's a bit different, in this function they create an object of their own rewardedAd type and call rewardedAd.render() which only calls initialize if the ad is not an RTB ad